kit/src/components/embeds/file_embed/mod.rs

```diff
@@ -365,3 +323,25 @@ fn get_file_thumbnail_if_is_image(filepath: PathBuf, filename: String) -> String
     };
     image
 }
+
+fn show_download_button_if_enabled<'a>(
+    cx: Scope<'a, Props<'a>>,
+    with_download_button: bool,
+    btn_icon: common::icons::outline::Shape,
+) -> Element<'a> {
+    if with_download_button {
```

Comment:
you might be able to do the following to simplify it

```rust
with_download_button.then_some(cx.render(rsx!(
            div {
                id: "file-embed-action-button",
                Button {
                    icon: btn_icon,
                    appearance: Appearance::Primary,
                    aria_label: "attachment-button".into(),
                    onpress: move |_| cx.props.on_press.call(()),
                }
            }
)))
```

so if its true, it will return `Element` (which is an alias of `Option`), otherwise it would return `None`.